subsys/net/ip/l2/ethernet.c

```diff
@@ -200,10 +200,15 @@ static enum net_verdict ethernet_send(struct net_if *iface,
 			return NET_DROP;
 		}
 
-		NET_DBG("Sending arp pkt %p (orig %p) to iface %p",
-			arp_pkt, pkt, iface);
-
-		pkt = arp_pkt;
+		if (pkt != arp_pkt) {
+			NET_DBG("Sending arp pkt %p (orig %p) to iface %p",
+				arp_pkt, pkt, iface);
+
```

Comment:
rebase your patch again latest master, some removed lines were re-introduced again.
besides that it's fine to me.